### PR TITLE
Add QNX to qualification report

### PR DESCRIPTION
**Pull request by Hoverbear**
Write the qualification docs for QNX. We only qualify with a specific QNX version for now.

Review:

```bash
./x.py doc ferrocene/doc/qualification-report --fresh
./x.py doc ferrocene/doc/safety-manual --fresh
./x.py doc ferrocene/doc/user-manual --fresh
```

### CODE REVIEW

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
The failure seems to have been transient. I [reran](https://app.circleci.com/pipelines/github/ferrocene/ferrocene/7040/workflows/739265f0-7fe9-4676-8943-e71dc64df6d1/jobs/39366) and the same failure did not happen.

**Comment by Hoverbear** [comment]
bors try

File: ferrocene/doc/safety-manual/src/scope.rst [hunk 1]
```diff
@@ -31,6 +31,14 @@ This qualification is restricted to the following environment:
      - :target:`x86_64-unknown-linux-gnu`
      - ``core``, ``alloc``, ``std``, ``proc_macro``, ``test``
 
+   * - :target:`x86_64-unknown-linux-gnu`
+     - :target:`x86_64-pc-nto-qnx710`
+     - ``core``, ``alloc``
+
+   * - :target:`x86_64-unknown-linux-gnu`
+     - :target:`aarch64-unknown-nto-qnx710`
+     - ``core``, ``alloc``
```
Comment (pietroalbini) [inline ferrocene/doc/safety-manual/src/scope.rst:40]: These targets also include `std`, `proc_macro`, and `test`.

**Review by pietroalbini — APPROVED** [review]
Only one nit.

File: ferrocene/doc/safety-manual/src/scope.rst (hunk 1 above)
Comment (Hoverbear) [reply ferrocene/doc/safety-manual/src/scope.rst:40]: Oh! Yes, good catch.

**Review by pietroalbini — APPROVED** [review]
bors merge

**Review by japaric — APPROVED** [review]

**Pull request merged**